llama-index-integrations/llms/llama-index-llms-cohere/llama_index/llms/cohere/base.py

```diff
@@ -131,13 +140,169 @@ def _get_all_kwargs(self, **kwargs: Any) -> Dict[str, Any]:
             **kwargs,
         }
 
+    def _prepare_chat_with_tools(
+        self,
+        tools: List["BaseTool"],
+        user_msg: Optional[Union[str, ChatMessage]] = None,
+        chat_history: Optional[List[ChatMessage]] = None,
+        verbose: bool = False,
+        allow_parallel_tool_calls: bool = False,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        """Prepare the chat with tools."""
+        chat_history = chat_history or []
+
+        if isinstance(user_msg, str):
+            user_msg = ChatMessage(role=MessageRole.USER, content=user_msg)
+            chat_history.append(user_msg)
+
+        tools_cohere_format = format_to_cohere_tools(tools)
+        return {
+            "messages": chat_history,
+            "tools": tools_cohere_format or [],
+            **kwargs,
+        }
+
+    def get_tool_calls_from_response(
+        self,
+        response: "ChatResponse",
+        error_on_no_tool_call: bool = True,
+    ) -> List[ToolSelection]:
+        """Predict and call the tool."""
+        tool_calls: List[ToolCall] = response.message.additional_kwargs.get(
+            "tool_calls"
+        )
+        if not tool_calls:
+            tool_calls = []
+
+        if len(tool_calls) < 1:
+            if error_on_no_tool_call:
+                raise ValueError(
+                    f"Expected at least one tool call, but got {len(tool_calls)} tool calls."
+                )
+            else:
+                return []
+
+        tool_selections = []
+        for tool_call in tool_calls:
+            if not isinstance(tool_call, ToolCall):
+                raise ValueError("Invalid tool_call object")
+            # argument_dict = json.loads(tool_call.parameters)
+            tool_selections.append(
+                ToolSelection(
+                    tool_id="TESTESTSE",
```

Comment:
why not pass the actual id?

Reply:
Oh shit this was a test id... I'll update haha thanks for catching!